lib/extension/dlisio/ext/io.hpp

```diff
@@ -25,61 +34,42 @@ struct record {
 
 class stream {
```

Comment:
The current implementation of `stream` basically wraps lfp functions and deals with errors before passing on the result. Main reason is to avoid the very verbose switch/case error handling from the lfp's C API. The question is if we are confident that we always make the right choice when handling errors internally in `stream`. There are two other alternatives as I see it:

1. Mirror lfp's API in `stream`. I.e. `stream` does no error handling leaving the caller to do the tedious switch/case. 
2. Register custom c++ exceptions (like we already have for io_error), and then map lfp's error codes to those exception. This way we don't have to deal with the nasty error codes outside of `stream`

Both solution moves the error-handling to the caller, meaning we have a better context of the current state when we decide how to deal with errors. However, I don't consider the first one to be a desirable option. I would probably opt for alternative 2 here, even though we have to do way more error-handling in the caller functions (`findfdata`, `extract`, `findoffsets`)

Reply:
OK, I see your point.
It might be a good idea to propagate error codes to get more context, but I am leaning more to modified option 1.

Here might come a question of division between io.cpp and dlisio.cpp
- dlisio.cpp operates on error codes
- methods in io.cpp deal with error codes, making sure they will not escape into the real world.

stream class unfortunately does escape into the outside world through core.
So that means that allowing stream just to pass the codes (your option 1) might leave them in the open, which is bad.
Option 2 will deal with this problem, but I do not like it as we will be forced to create either 1x1 mapping of all possible lfp exceptions, or combine them somehow. Also it would introduce some try/catch to friend io.cpp methods, which looks to go against current design.

So in my opinion we should
1. make sure that only the private stream methods are allowed to return error codes
or
2. get rid of them completely by moving all the methods that return error codes to dlisio.cpp